test/integration/router_test.go

```diff
@@ -1048,16 +1129,25 @@ func getRoute(routerUrl string, hostName string, protocol string, headers map[st
 		if protocol == "wss" {
 			port = 443
 		}
-		wsConfig.Location.Host = fmt.Sprintf("%s:%d", hostName, port)
-		wsConfig.TlsConfig = tlsConfig
+		if strings.Contains(hostName, ":") {
+			wsConfig.Location.Host = hostName
+		} else {
+			wsConfig.Location.Host = fmt.Sprintf("%s:%d", hostName, port)
+		}
 
-		ws, err := websocket.DialConfig(wsConfig)
+		var conn net.Conn
+		if tlsConfig == nil {
+			conn, err = dialer("tcp", wsConfig.Location.Host)
+		} else {
+			conn, err = tlsDialer("tcp", wsConfig.Location.Host, tlsConfig)
+		}
+		if err != nil {
+			return "", err
+		}
+		ws, err := websocket.NewClient(wsConfig, conn)
 		if err != nil {
-			if derr, ok := err.(*websocket.DialError); ok {
-				if derr.Err == websocket.ErrBadStatus {
-					// a better websocket library would know the difference here
-					return "", ErrUnavailable
-				}
+			if err == websocket.ErrBadStatus {
```

Comment:
the dial error case would have been encountered a few lines above, during the dial